commit 90c0f8676b06041c599fd941514e09ba8704f0ef
Author: Martin Pitt <martin.pitt@ubuntu.com>
Date:   Tue Feb 22 08:35:37 2011 +0100

    automount: Fix crash when unlocking screen saver
    
    Fix commit 71deedf: We _do_ expect that some volumes have queued up in
    check_volume_queue() once the screen saver got inactive, so don't crash with an
    assertion failure in this case.
    
    Just check whether the screen saver is active, and don't mount anything if so.

diff --git a/plugins/automount/gsd-automount-manager.c b/plugins/automount/gsd-automount-manager.c
index 788a5fe..5988f2b 100644
--- a/plugins/automount/gsd-automount-manager.c
+++ b/plugins/automount/gsd-automount-manager.c
@@ -184,8 +184,8 @@ check_volume_queue (GsdAutomountManager *manager)
 
         l = manager->priv->volume_queue;
 
-        if (!manager->priv->screensaver_active) {
-                g_assert (l == NULL);
+        if (manager->priv->screensaver_active) {
+                return;
         }
 
         while (l != NULL) {
